docs/asciidoc/http-client.adoc

```diff
@@ -512,3 +518,188 @@ include::{examplesdir}/resolver/custom/Application.java[lines=18..38]
+if `SO_KEEPALIVE` has been set. The maximum time is implementation dependent, but typically the default is 2 hours.
+** `TCP_KEEPINTVL` - the time (resolution: seconds) between individual `keepalive` probes.
+** `TCP_KEEPCNT` - the maximum number of `keepalive` probes TCP should send before dropping the connection.
+
+NOTE: Sometimes between the client and the server you may have a network component that silently drops the idle connections without sending a response.
+From Reactor Netty point of view in this use case the remote peer just does not respond.
+In order to be able to handle such use case you may consider configuring
+https://docs.oracle.com/javase/8/docs/technotes/guides/net/socketOpt.html[SO_KEEPALIVE].
+
+To customize the default settings, you can configure `HttpClient` as follows:
+
+====
+[source,java,indent=0]
+.{examplesdir}/channeloptions/Application.java
+----
+include::{examplesdir}/channeloptions/Application.java[lines=18..46]
+----
+<1> Configures the connection establishment timeout to 10 seconds.
+<2> Enables TCP `keepalive`. This means that TCP starts sending `keepalive` probes when a connection is idle for some time.
+<3> The connection needs to remain idle 5 minutes before TCP starts sending `keepalive` probes.
+<4> Configures the time between individual `keepalive` probes to 1 minute.
+<5> Configures the maximum number of TCP `keepalive` probes to 8.
+====
+
+[[ssl-tls-timeout]]
+==== SSL/TLS Timeout
+`HttpClient` supports the SSL/TLS functionality provided by Netty.
+
+The below shows the available timeout configuration options:
+
+* `handshakeTimeout` - this option can be used for configuring SSL handshake timeout (resolution: ms). Default: 10s.
+
+NOTE: You should consider increasing the SSL handshake timeout when expecting slow network connections.
+
+* `closeNotifyFlushTimeout` - this option can be used for configuring SSL `close_notify` flush timeout (resolution: ms). Default: 3s.
+* `closeNotifyReadTimeout` - this option can be used for configuring SSL `close_notify` read timeout (resolution: ms). Default: 0s.
+
+To customize the default settings, you can configure `HttpClient` as follows:
+
+====
+[source,java,indent=0]
+.{examplesdir}/security/custom/Application.java
+----
+include::{examplesdir}/security/custom/Application.java[lines=18..47]
+----
+<1> Configures the SSL handshake timeout to 30 seconds.
+<2> Configures the SSL `close_notify` flush timeout to 10 seconds.
+<3> Configures the SSL `close_notify` read timeout to 10 seconds.
+====
+
+[[proxy-timeout]]
+==== Proxy Timeout
+`HttpClient` supports the proxy functionality provided by Netty and provides a way to specify
+{javadoc}/reactor/netty/transport/ProxyProvider.Builder.html#connectTimeoutMillis-long-[connection establishment timeout].
+If the connection establishment attempt to the remote peer does not finish within the timeout,
+the connection establishment attempt will be failed. Default: 10s.
```

Comment:
Change "will be failed" to "fails." 

(A general rule is to always use the indefinite present tense - sometimes called the simple present tense - unless there's no way to avoid doing otherwise. As a result, our docs rarely have "will".)